MekHQ/src/mekhq/campaign/force/Force.java

```diff
@@ -197,12 +213,19 @@ public void addSubForce(Force sub, boolean assignParent) {
     }
 
     /**
+     * @param combatForcesOnly to only include combat forces or to also include non-combat forces
      * @return all the unit ids in this force and all of its subforces
      */
-    public Vector<UUID> getAllUnits() {
-        Vector<UUID> allUnits = new Vector<>(units);
+    public Vector<UUID> getAllUnits(boolean combatForcesOnly) {
```

Comment:
This is where your dislike of overloads comes back to bite you in the ass, this PR would have been about five instead of twenty files, with a lot less diffs if you keep the existing method and default that value to true.

Reply:
I'd far rather have this in the future to improve readability. More work now means less work in the future, which is a trade I'd take any time.

Reply:
a) I'm not convinced that readability is improved in any way here (it's very much in the eye of the beholder) and b) the surface area of code changes is greater, which means a greater chance of introducing defects.

Reply:
The other side is by doing this I actually caught a few bugs that would have otherwise gone unseen, because I needed to go through every single call to the method to verify it was using the correct true/false value.

Reply:
However, I have really bloated my recent PRs with minor formatting fixes and what not while not doing them in separate commits, which I apologize for.

Reply:
In this case, it's not really worth it to roll all that work back anyway.